### DISCUSSION TITLE
Pico with MHS 3.5 TFT (ILI9486/XPT2046)

### DISCUSSION THREAD

**Discussion by DanielYWoo** [Bodmer/TFT_eSPI#3087] [Raspberry Pi Pico]
I have a 3.5 TFT with ILI9486/XPT2046 (http://www.lcdwiki.com/MHS-3.5inch_RPi_Display), just like the image in your doc:
https://github.com/Bodmer/TFT_eSPI/blob/master/docs/RPi_TFT_connections/RPi_TFT_Connections.png

This is designed for Raspberry 2/3/4 but it should work with Pico in Arduino IDE since the ILI9486 and XPT2046 are pretty widely used. My test code is as below

```
#define ILI9486_DRIVER
#define TFT_MISO  12
#define TFT_MOSI  11
#define TFT_SCLK  10
#define TFT_CS     9
#define TFT_DC     8
#define TFT_RST   15
#define TOUCH_CS 21

#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT

#define TFT_SPI_PORT 1 // Set to 0 if SPI0 pins are used, or 1 if spi1 pins used

#include <TFT_eSPI.h> // Hardware-specific library
#include <SPI.h>

TFT_eSPI tft = TFT_eSPI();       // Invoke custom library

void setup(void) {
  tft.init();
  tft.setRotation(1);
}

void loop() {

  tft.fillScreen(TFT_BLACK);

  // Draw some random ellipses
  for (int i = 0; i < 40; i++)
  {
    int rx = random(60);
    int ry = random(60);
    int x = rx + random(320 - rx - rx);
    int y = ry + random(240 - ry - ry);
    tft.fillEllipse(x, y, rx, ry, random(0xFFFF));
  }

  delay(2000);
  tft.fillScreen(TFT_BLACK);

  for (int i = 0; i < 40; i++)
  {
    int rx = random(60);
    int ry = random(60);
    int x = rx + random(480 - rx - rx);
    int y = ry + random(320 - ry - ry);
    tft.drawEllipse(x, y, rx, ry, random(0xFFFF));
  }

  delay(2000);
}

```

The wiring is 

```

TFT 5v - PICO vbus
TFT 3.3v - PICO 3.3v
TFT GND - PICO GND
TFT_MISO  - GP12
TFT_MOSI  - GP11
TFT_SCLK  - GP10
TFT_CS  - GP9
TFT_DC -  GP8
TFT_RST - GP15
TOUCH_CS - GP21
```

However, the screen is on but nothing shows, any clue to troubleshoot this?

**Comment by brightproject** [comment]
> the screen is on but nothing shows

Try this library [Mcufriend_kbv](https://github.com/prenticedavid/MCUFRIEND_kbv)
And especially a [test](https://github.com/prenticedavid/MCUFRIEND_kbv/blob/master/examples/LCD_ID_readnew/LCD_ID_readnew.ino) sketch to make sure that you have connected the pins correctly.
My ILI9486 display work [fine](https://github.com/Bodmer/TFT_eSPI/issues/2682#issuecomment-1576552075).
